pkg/sink/sink.go

```diff
@@ -53,25 +67,44 @@ func (r Resource) HandleEvent(response http.ResponseWriter, request *http.Reques
 		return
 	}
 
+	var wg sync.WaitGroup
+	wg.Add(len(el.Spec.Triggers))
+	header := request.Header
+
 	// Execute each Trigger
 	for _, trigger := range el.Spec.Triggers {
-		binding, err := template.ResolveBinding(trigger,
-			r.TriggersClient.TektonV1alpha1().TriggerBindings(r.EventListenerNamespace).Get,
-			r.TriggersClient.TektonV1alpha1().TriggerTemplates(r.EventListenerNamespace).Get)
-		if err != nil {
-			log.Print(err)
-			continue
-		}
-		resources, err := template.NewResources(event, binding)
-		if err != nil {
-			log.Print(err)
-			continue
-		}
-		err = createResources(resources, r.RESTClient, r.DiscoveryClient)
-		if err != nil {
-			log.Print(err)
+		go r.executeTrigger(event, header, trigger, &wg)
+	}
+	wg.Wait()
```

Comment:
Why do we need to wait for each trigger to finish before we return from the handleEvent function? Can it not just call executeTrigger for each trigger and then return?

Reply:
It does make more sense. The only reason I did that was because of the advice that I read that "Never start a goroutine without knowing how it will stop".  I feel we should quickly try to respond to the request but at the same time, I feel we should log somewhere that request came.

Reply:
That seems like a good approach. Given that we retry for a fixed number of times in the goroutine, I'd argue that we know when it'd stop :smile: And yeah, responding to the request quickly was the main concern. 
In the far off future, we'd probably have to accept the event, queue it up for processing, and then process events off a queue!